apps/dashboard/src/app/(dashboard)/dashboard/connect/analytics/page.tsx

```diff
@@ -0,0 +1,28 @@
+import { redirect } from "next/navigation";
+import { getApiKeys } from "../../../../api/lib/getAPIKeys";
+import { getAuthToken } from "../../../../api/lib/getAuthToken";
+import { ConnectAnalyticsPage } from "./ConnectAnalyticsPage";
+
+export default async function Page(props: {
+  searchParams: {
+    clientId?: string;
+  };
+}) {
+  const authToken = getAuthToken();
+  const { clientId } = props.searchParams;
+
+  if (!authToken) {
+    const params = new URLSearchParams();
+    params.append("next", "/dashboard/connect/analytics");
+    if (clientId) {
+      params.append("clientId", clientId);
+    }
+    redirect(`/login?${params.toString()}`);
+  }
+
+  const apiKeys = await getApiKeys();
+
+  return <ConnectAnalyticsPage defaultClientId={clientId} apiKeys={apiKeys} />;
+}
+
+export const dynamic = "force-dynamic";
```

Comment:
is this required? next should behave like this automatically since we're accessing `cookies()`?

Reply:
next should make this page dynamic, but we've seen errors in sentry earlier where next was not making the page dynamic even when the cookies() was used. so adding this explicitly

Reply:
deleted